x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/security/authz_api_keys/40_document_level_security_synthetic_source.yml

```diff
@@ -0,0 +1,386 @@
+---
+setup:
+  - skip:
+      features: headers
+
+  - do:
+      cluster.health:
+        wait_for_status: yellow
+
+---
+Filter on single field:
```

Comment:
I see these all all `search` api calls...maybe we also need to test getting a document by id checking not accessible fields are not available in the response?

Reply:
This is document-level security so it applies at the document level - no fields get hidden: 
https://www.elastic.co/guide/en/elasticsearch/reference/current/document-level-security.html

I think it's important to verify that scripts can access source contentes as expected, the rest seem orthogonal to synthetic source.

Reply:
Right, this would be handled by field level security. I don't think we need to test this here.

Reply:
Well, it doesn't hurt to include a test for this. Added one.